src/main/java/uk/gov/hmcts/reform/document/DocumentUploadClientApi.java

```diff
@@ -47,14 +50,34 @@ public DocumentUploadClientApi(
         this.objectMapper = objectMapper;
     }
 
-    public UploadResponse upload(
+    public UploadResponse uploadWithUserId(
         @RequestHeader(HttpHeaders.AUTHORIZATION) String authorisation,
         @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuth,
         @RequestHeader(USER_ID) String userId,
         @RequestPart List<MultipartFile> files
+    ) {
+        return upload(authorisation, serviceAuth, userId, Collections.emptyList(), files);
+    }
+
+    public UploadResponse uploadWithRoles(
+        @RequestHeader(HttpHeaders.AUTHORIZATION) String authorisation,
+        @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuth,
+        @RequestHeader(USER_ID) String userId,
```

Comment:
do you need user id and roles?

this is more just the upload method I think?

Reply:
yes it uses userId  in auditEvents to mark read and create of documents